src/wat-writer.cc

```diff
@@ -464,6 +490,8 @@ void WatWriter::WriteConst(const Const* const_) {
 }
 
 void WatWriter::WriteExpr(const Expr* expr) {
+  if (s_trace)
```

Comment:
Ok, I've simplified the tracing so that it is minimized.

I created files tracing.{h,cc} that implement a simple (scoped) class that generates enter/exit messages automatically. I then added two simple macros WABT_TRACE(method_name) and WABT_TRACE_ARGS(method_name, format, ...) to insert an instance of this (scoped) tracing class.